WordPress/Classes/ViewRelated/Reader/Select Interests/ReaderSelectInterestsViewController.swift

```diff
@@ -321,7 +321,10 @@ extension ReaderSelectInterestsViewController: UICollectionViewDataSource {
             fatalError("Expected a ReaderInterestsCollectionViewCell for identifier: \(Constants.reuseIdentifier)")
         }
 
-        let interest: ReaderInterestViewModel = dataSource.interest(for: indexPath.row)
+        guard let interest = dataSource.interest(for: indexPath.row) else {
+            DDLogError("ReaderSelectInterestsViewController: Requested for data at invalid row \(indexPath.row)")
```

Comment:
[Discussion] What do you think of reporting this error to Sentry instead of silently handling it? I think an invalid `indexPath.row` is unexpected, so this crash could be a symptom of a deeper underlying problem. 

```swift
CrashLogging.main.logMessage("ReaderSelectInterestsViewController: Requested for data at invalid row", properties: ["row": indexPath.row], level: .warning)
```

Reply:
Sounds good to me. Addressed in https://github.com/wordpress-mobile/WordPress-iOS/pull/22876/commits/42ab6a8241cae55ba5e6860a498e5c744ce8854b!